tests/transforms/test_generate_heatmap.py

```diff
@@ -0,0 +1,312 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+import unittest
+
+import numpy as np
+import torch
+from parameterized import parameterized
+
+from monai.transforms.post.array import GenerateHeatmap
+from tests.test_utils import TEST_NDARRAYS
+
+
+def _argmax_nd(x) -> np.ndarray:
+    """argmax for N-D array → returns coordinate vector (z,y,x) or (y,x)."""
+    if isinstance(x, torch.Tensor):
+        x = x.cpu().numpy()
+    return np.asarray(np.unravel_index(np.argmax(x), x.shape))
+
+
+# Test cases for 2D array inputs with different data types
+TEST_CASES_2D = []
+for idx, p in enumerate(TEST_NDARRAYS):
+    TEST_CASES_2D.append(
+        [
+            f"2d_basic_type{idx}",
+            p(np.array([[4.2, 7.8], [12.3, 3.6]], dtype=np.float32)),
+            {"sigma": 1.5, "spatial_shape": (16, 16)},
+            (2, 16, 16),
+        ]
+    )
```

Comment:
```suggestion
TEST_CASES_2D = [
    [
        f"2d_basic_type{idx}",
        p(np.array([[4.2, 7.8], [12.3, 3.6]], dtype=np.float32)),
        {"sigma": 1.5, "spatial_shape": (16, 16)},
        (2, 16, 16),
    ]
    for idx, p in enumerate(TEST_NDARRAYS)
]
```
If you wanted I think defining values like this is fine.